### PR TITLE
[dart][dart-dio] Add support for default values

### PR DESCRIPTION
**Pull request by kuhnroyal**
This PR adds support for default values to the dart-dio generator.

* return `null` when there is no default value instead of `"null"` (this also removes the "defaults to null" from model docs)
* don't need any collection literal default values for built_value
* use built_value's `_initializeBuilder` to set default values
*  fix formatting of default values in model doc

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

**Comment by agilob** [comment]
>return null when there is no default value instead of "null" (this also removes the "defaults to null" from model docs)

:+1:

**Review by agilob — APPROVED** [review]

**Review by agilob — APPROVED** [review]

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java [hunk 1]
```diff
@@ -125,20 +125,13 @@ public String getHelp() {
 
     @Override
     public String toDefaultValue(Schema schema) {
-        if (ModelUtils.isMapSchema(schema)) {
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:128]: Does it make sense to remove these 4 lines from `DartClient` too?

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:128]: No, in the dart generator you need the collection default values. But in dart-dio there are none, we use a builder pattern there.

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:128]: The models are mutable and you said you would expect to have an empty collection that you can just add to after instantiating the object. This is what the collection default values are used for.

Comment (agilob) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:128]: >This is what the collection default values are used for.

Yes, when I worked on that I kept it for backwards compatibility, I would do it differently now: keep compatibility with most popular generators where default value is null, unless field is required or defined as non-null. At some point we'll have to break it, but not rush.

**Pull request merged**